skada/_subspace.py

```diff
@@ -366,3 +366,233 @@ def TransferComponentAnalysis(
         ),
         base_estimator,
     )
+
+
+class TJMAdapter(BaseAdapter):
+    """Domain Adaptation Using TJM: Transfer Joint .
+
+    See [1]_ for details.
+
+    Parameters
+    ----------
+    base_estimator : object, default=None
+        estimator used for fitting and prediction
+    kernel : kernel object, default='rbf'
+        The kernel computed between data.
+    n_components : int, default=None
+        The numbers of components to learn with PCA.
+        Should be less or equal to the number of samples
+        of the source and target data.
+
+    Attributes
+    ----------
+    None
+
+    References
+    ----------
+    .. [1]
+        [Long et al., 2014] Long, M., Wang, J., Ding, G., Sun, J., and Yu, P.
+        (2014). Transfer joint matching for unsupervised domain adaptation.
+        In IEEE Conference on Computer Vision and Pattern Recognition (CVPR),
+        pages 1410–1417
+
+    """
+
+    def __init__(
+        self,
+        n_components=None,
+        random_state=None,
+        regularizer=0,
+        max_iter=100,
+        kernel="rbf",
+        tol=0.05,
+    ):
+        super().__init__()
+        self.n_components = n_components
+        self.regularizer = regularizer
+        self.random_state = random_state
+        self.kernel = kernel
+        self.max_iter = max_iter
+        self.tol = tol
+
+    def adapt(self, X, y=None, sample_domain=None, **kwargs):
+        """Predict adaptation (weigts, sample or labels).
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The source data.
+        y : array-like, shape (n_samples,)
+            The source labels.
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels (same as sample_domain).
+
+        Returns
+        -------
+        X_t : array-like, shape (n_samples, n_components)
+            The data transformed to the target subspace.
+        y_t : array-like, shape (n_samples,)
+            The labels (same as y).
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels transformed to the target subspace
+            (same as sample_domain).
+        weigts : None
+            No weigts are returned here.
```

Comment:
these return statements seem wrong

Reply:
yes, I just placed then so that pre-commit was not preventing me from committing, I had a problem with the algorithm and wanted @tgnassou to be able to see the algorithm, I will correct those now